dialogy/plugins/text/classification/xlmr.py

```diff
@@ -176,13 +192,36 @@ def inference(
             raise ValueError(
                 f"Plugin {self.__class__.__name__} requires state to be passed to the model."
             )
-        elif self.use_state and state:
-            texts = [f"{text} <s> {state} </s>" for text in texts]
+
+        if self.use_prompt and not state:
+            raise ValueError(
+                f"In order to use prompts as feature, Plugin {self.__class__.__name__} is requires state to be passed to the model."
+            )
+
+        if self.use_prompt and not lang:
+            raise ValueError(
+                f"In order to use prompts as feature, Plugin {self.__class__.__name__} is requires lang to be passed to the model."
+            )
+
+        if self.use_prompt and state:
+            texts[0] += "<s> " + self.lookup_prompt(lang=lang, state=state)[0] + " </s>"
+            if self.debug:
```

Comment:
I don't think you should be needing an if construct here if logging level is properly configured while running the framework.